### PR TITLE
PR. Some restrictions for methods

### LINKED ISSUE #21: Some restrictions for methods
Add new field “ownerId” to the “Project” entity. This field should be filled during project creation and should be never updated.
The following API should be available for the project owner only:
•	PUT /api/projects/{id}
•	DELETE /api/projects/{id}
Otherwise, server should return 403 error code.

Start after #20

### PR DESCRIPTION
**Pull request by tempo92**
Closes #21

### CODE REVIEW

File: src/main/java/org/tempo/springEdu/service/ProjectService.java [hunk 1]
```diff
@@ -26,14 +30,24 @@
...
+        }
     }
 
     public void delete(String id) {
         Project delProject = findById(id);
-        repository.delete(delProject);
+        if (getUserId().equals(delProject.getOwnerId())) {
+            repository.delete(delProject);
+        } else {
+            throw new AccessDeniedException("Owner only");
```
Comment (iv21dk) [inline src/main/java/org/tempo/springEdu/service/ProjectService.java:49]: Please try to resolve the issue using `@PreAuthorize` annotation and custom `PermissionEvaluator`

File: src/main/java/org/tempo/springEdu/service/ProjectService.java [hunk 2]
```diff
@@ -95,4 +112,9 @@ private Project dtoToEntity(ProjectDto dto) {
     private Project dtoToEntity(ProjectUpdateDto dto) {
         return modelMapper.map(dto, Project.class);
     }
+
+    private String getUserId() {
```
Comment (iv21dk) [inline src/main/java/org/tempo/springEdu/service/ProjectService.java:116]: Please try to use `@AuthenticationPrincipal` instead

**Review by iv21dk — CHANGES_REQUESTED** [review]

File: src/main/java/org/tempo/springEdu/service/ProjectService.java (hunk 1 above)
Comment (tempo92) [reply src/main/java/org/tempo/springEdu/service/ProjectService.java]: Fixed

File: src/main/java/org/tempo/springEdu/service/ProjectService.java (hunk 2 above)
Comment (tempo92) [reply src/main/java/org/tempo/springEdu/service/ProjectService.java]: Fixed

File: src/main/java/org/tempo/springEdu/security/CustomPermissionEvaluator.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import org.tempo.springEdu.entity.User;
+import org.tempo.springEdu.exception.ObjectNotFoundException;
+import org.tempo.springEdu.service.ProjectService;
+
+import java.io.Serializable;
+
+@Component
+public class CustomPermissionEvaluator implements PermissionEvaluator {
+
+    private final Logger logger = LogManager.getLogger(this.getClass());
```
Comment (iv21dk) [inline src/main/java/org/tempo/springEdu/security/CustomPermissionEvaluator.java:19]: You can use `@Log4j2` annotation instead. But in this case you will need to use `log` variable

File: src/main/java/org/tempo/springEdu/security/CustomPermissionEvaluator.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    private ProjectService projectService;
+
+    @Override
+    public boolean hasPermission(
+            Authentication authentication, Object targetDomainObject, Object permission) {
+        var projectId = targetDomainObject.toString();
+        var userId = ((User) authentication.getPrincipal()).getId();
+
+        logger.debug("hasPermission test");
+        logger.debug(
```
Comment (iv21dk) [inline src/main/java/org/tempo/springEdu/security/CustomPermissionEvaluator.java:31]: You can put your parameters using `{}` marker. So, the line can be like this:
```java
logger.debug("\nuserName: {}; userId: {}; projectId: {}", authentication.getName(), userId, projectId);
```

**Review by iv21dk — CHANGES_REQUESTED** [review]

**Review by iv21dk — APPROVED** [review]

File: src/main/java/org/tempo/springEdu/security/CustomPermissionEvaluator.java (hunk 4 above)
Comment (tempo92) [reply src/main/java/org/tempo/springEdu/security/CustomPermissionEvaluator.java:31]: I`ll keep that in mind

File: src/main/java/org/tempo/springEdu/security/CustomPermissionEvaluator.java (hunk 3 above)
Comment (tempo92) [reply src/main/java/org/tempo/springEdu/security/CustomPermissionEvaluator.java:19]: I'll keep that in mind

**Pull request merged**